### DISCUSSION TITLE
Session processing not running

### DISCUSSION THREAD

**Discussion by asarnow** [nextpyp/.github#22] [Q&A] [answered]
Offline processing is working great for me on our SLURM cluster, however session pre-processing is failing immediately with very little logging.

This is the log from the File Transfer step:
```
2023-12-04 15:49:22 [INFO] pyp/system/db_comm.py:34 | Parameters entered into database successfully
2023-12-04 15:49:22 [INFO] pyp/system/local_run.py:65 | chmod u+x /data/veesler/nextpyp/shared/sessions/epCgTnAHWqFeOH3R/Veesler-ep3J0d3dysqxHGgz/test2-8ngM7U2/20231204_154922_Veesler-ep3J0d3dysqxHGgz_test2-8ngM7U2_daemon.swarm
2023-12-04 15:49:23 [INFO] /opt/pyp/bin/run/streampyp:284 | /data/veesler/nextpyp/shared/sessions/epCgTnAHWqFeOH3R/Veesler-ep3J0d3dysqxHGgz/test2-8ngM7U2/20231204_154922_Veesler-ep3J0d3dysqxHGgz_test2-8ngM7U2_daemon.swarm not recognized as an image type, skipping
2023-12-04 15:49:23 [INFO] pyp/system/slurm.py:417 | Submitting 1 job(s) (epCqCn0Ls0PoVpwc)
2023-12-04 15:49:24 [INFO] /opt/pyp/bin/run/streampyp:643 | Entering loop with timeout = 5 day(s)
```

There is no output in the pre-processing job log, however the nextPYP server log for the job has the line:
`slurmstepd: error: Failed to invoke task plugins: task_p_pre_launch error`

So far I've been unable to determine the issue, does anyone have any suggestion? Is the "not recognized as an image type" message relevant? Thanks!

**Comment by abartesaghi** [comment]
Can you share the actual `streampyp` command at the top of the log from the `File Transfer` daemon?

**Comment by asarnow** [comment]
The command is:
`"/opt/pyp/bin/run/streampyp" -scope_pixel 1.37 -scope_voltage 300.0 -scope_dose_rate 3.3 -scope_tilt_axis 175.0 -data_path "/data/veesler/serialem/asarnow/test/*.tif" -data_mode tomo -gain_reference "/data/veesler/serialem/asarnow/23dec02b/CountCDSRef_u2os-64kx_648_0000_-0.0.mrc" -ctf_min_def 5000.0 -tomo_ali_patches 5 -tomo_rec_thickness 3072 -stream_session_name "test2-8ngM7U2" -stream_session_group "Veesler-ep3J0d3dysqxHGgz" -stream_transfer_fileset "*.mdoc" -stream_num_tilts 41 -stream_transfer_target "/data/veesler/nextpyp/shared/sessions/epCgTnAHWqFeOH3R" -stream_transfer_operation link -slurm_verbose -slurm_tasks 4 -slurm_daemon_tasks 4`

And the console output has the additional line:
`Matplotlib created a temporary cache directory at /tmp/matplotlib-enj3vf_z because the default path (/home_local/nextpyp/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.`

**Comment by abartesaghi** [comment, marked as answer]
The only thing unusual is slurm's error message `slurmstepd: error: Failed to invoke task plugins: task_p_pre_launch error` which means that slurm failed to start the pre-processing daemon. We haven't seen this error before, but there seem to be some [bug reports](https://bugs.schedmd.com/show_bug.cgi?id=10460) that could point to an issue with older versions of slurm. We don't understand why you are seeing this on the sessions side and not on the projects side though. I'm assuming you made sure the `/data/veesler/serialem/` and `/data/veesler/nextpyp` shares are visible from the compute nodes, right?

Regarding the options you are using to start the session, if you want to retrieve metadata from .mdoc files you should be using version [0.5.2](https://nextpyp.app/files/pyp/0.5.2/docs/index.html) or later, and you should specify the absolute path to the .mdoc files in the `Location of mdoc files` field (`Raw data` tab)  and check the option `Read metadata from mdoc files` in the Frame alignment tab.

**Reply by abartesaghi** [reply]
Since we haven't heard back from you, we assume this issue was resolved, but please let us know otherwise.

**Reply by asarnow** [reply]
Sorry Alberto, we still haven't figured it out but I am 99.9% certain it's not related to PYP. It might be something about the slurm daemon on some types of CPUs.
